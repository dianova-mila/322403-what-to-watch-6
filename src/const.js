const ALL_GENRES = `All genres`;

const Tabs = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`,
};

export {ALL_GENRES, Tabs};
