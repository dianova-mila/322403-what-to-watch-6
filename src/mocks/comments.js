export default [
  {
    "id": 1,
    "user": {
      "id": 10,
      "name": `Max`
    },
    "rating": 3.9,
    "comment": `Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.`,
    "date": `2021-02-18T08:04:28.658Z`
  },
  {
    "id": 2,
    "user": {
      "id": 19,
      "name": `Christina`
    },
    "rating": 9.7,
    "comment": `It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.`,
    "date": `2021-02-17T08:04:28.658Z`
  }
];
