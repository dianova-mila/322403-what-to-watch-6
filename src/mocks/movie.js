export default {
  name: `Macbeth`,
  posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg`,
  backgroundColor: `#F1E9CE`,
  description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
  rating: 3.3,
  scoresCount: 48798,
  director: `Justin Kurzel`,
  starring: [
    `Michael Fassbender`,
    `Marion Cotillard`,
    `Jack Madigan`
  ],
  runTime: 113,
  genre: `Drama`,
  released: 2015,
  id: `1`,
  isFavorite: false,
  videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
};
