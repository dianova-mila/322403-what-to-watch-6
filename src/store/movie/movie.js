import {createReducer} from "@reduxjs/toolkit";
import {loadOneMovie, loadPromoMovie, loadComments} from "../actions";

const initialState = {
  movie: {},
  promoMovie: {},
  isPromoMovieLoaded: false,
  comments: [],
  isOneMovieLoaded: false,
};

const movie = createReducer(initialState, (builder) => {
  builder.addCase(loadOneMovie, (state, action) => {
    state.movie = action.payload;
    state.isOneMovieLoaded = true;
  });

  builder.addCase(loadPromoMovie, (state, action) => {
    state.promoMovie = action.payload;
    state.isPromoMovieLoaded = true;
  });

  builder.addCase(loadComments, (state, action) => {
    state.comments = action.payload;
  });
});

export {movie};
