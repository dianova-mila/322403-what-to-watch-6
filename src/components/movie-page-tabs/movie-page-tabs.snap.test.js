import React from "react";
import {render, screen} from "@testing-library/react";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";
import configureStore from 'redux-mock-store';
import {createMemoryHistory} from 'history';
import userEvent from '@testing-library/user-event';
import movie from "../../mocks/movie";
import comments from "../../mocks/comments";
import MoviePageTabs from "./movie-page-tabs";
import {Tabs} from "../../const";

describe(`MoviePageTabs should render correctly`, () => {
  it(`MoviePageTabs should render correctly, when selected Overview`, () => {
    const {container} = render(<MoviePageTabs defaultTab={Tabs.OVERVIEW} movie={movie} comments={comments}/>);

    expect(container).toMatchSnapshot();
  });

  it(`MoviePageTabs should render correctly, when selected Details`, () => {
    const {container} = render(<MoviePageTabs defaultTab={Tabs.OVERVIEW} movie={movie} comments={comments}/>);

    userEvent.click(screen.getByText(/Details/i));

    expect(container).toMatchSnapshot();
  });

  it(`MoviePageTabs should render correctly, when selected Reviews`, () => {
    const {container} = render(<MoviePageTabs defaultTab={Tabs.OVERVIEW} movie={movie} comments={comments}/>);

    userEvent.click(screen.getByText(/Reviews/i));

    expect(container).toMatchSnapshot();
  });

  it(`MoviePageTabs should render correctly in AddReview page`, () => {
    const mockStore = configureStore({});
    const history = createMemoryHistory();

    const {container} = render(
        <Provider store={mockStore({})}>
          <Router history={history}>
            <MoviePageTabs defaultTab={Tabs.REVIEWS} movie={movie} comments={comments}/>
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
