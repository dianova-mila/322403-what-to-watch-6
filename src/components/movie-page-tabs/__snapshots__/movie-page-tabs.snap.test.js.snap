// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePageTabs should render correctly MoviePageTabs should render correctly in AddReview page 1`] = `
<div>
  <div
    class="movie-card__desc"
  >
    <nav
      class="movie-nav movie-card__nav"
    >
      <ul
        class="movie-nav__list"
      >
        <li
          class="movie-nav__item "
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Overview
          </a>
        </li>
        <li
          class="movie-nav__item "
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Details
          </a>
        </li>
        <li
          class="movie-nav__item movie-nav__item--active"
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Reviews
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="add-review"
    >
      
      <form
        action="#"
        class="add-review__form"
      >
        <div
          class="rating"
        >
          <div
            class="rating__stars"
          >
            <input
              class="rating__input"
              id="star-1"
              name="rating"
              type="radio"
              value="1"
            />
            <label
              class="rating__label"
              for="star-1"
            >
              Rating 1
            </label>
            <input
              class="rating__input"
              id="star-2"
              name="rating"
              type="radio"
              value="2"
            />
            <label
              class="rating__label"
              for="star-2"
            >
              Rating 2
            </label>
            <input
              class="rating__input"
              id="star-3"
              name="rating"
              type="radio"
              value="3"
            />
            <label
              class="rating__label"
              for="star-3"
            >
              Rating 3
            </label>
            <input
              class="rating__input"
              id="star-4"
              name="rating"
              type="radio"
              value="4"
            />
            <label
              class="rating__label"
              for="star-4"
            >
              Rating 4
            </label>
            <input
              class="rating__input"
              id="star-5"
              name="rating"
              type="radio"
              value="5"
            />
            <label
              class="rating__label"
              for="star-5"
            >
              Rating 5
            </label>
            <input
              class="rating__input"
              id="star-6"
              name="rating"
              type="radio"
              value="6"
            />
            <label
              class="rating__label"
              for="star-6"
            >
              Rating 6
            </label>
            <input
              class="rating__input"
              id="star-7"
              name="rating"
              type="radio"
              value="7"
            />
            <label
              class="rating__label"
              for="star-7"
            >
              Rating 7
            </label>
            <input
              class="rating__input"
              id="star-8"
              name="rating"
              type="radio"
              value="8"
            />
            <label
              class="rating__label"
              for="star-8"
            >
              Rating 8
            </label>
            <input
              class="rating__input"
              id="star-9"
              name="rating"
              type="radio"
              value="9"
            />
            <label
              class="rating__label"
              for="star-9"
            >
              Rating 9
            </label>
            <input
              class="rating__input"
              id="star-10"
              name="rating"
              type="radio"
              value="10"
            />
            <label
              class="rating__label"
              for="star-10"
            >
              Rating 10
            </label>
          </div>
        </div>
        <div
          class="add-review__text"
        >
          <textarea
            class="add-review__textarea"
            id="review-text"
            maxlength="400"
            minlength="50"
            name="review-text"
            placeholder="Review text"
            required=""
          />
          <div
            class="add-review__submit"
          >
            <button
              class="add-review__btn"
              disabled=""
              type="submit"
            >
              Post
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      class="movie-card__reviews movie-card__row"
    >
      <div
        class="movie-card__reviews-col"
      >
        <div
          class="review"
        >
          <blockquote
            class="review__quote"
          >
            <p
              class="review__text"
            >
              Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.
            </p>
            <footer
              class="review__details"
            >
              <cite
                class="review__author"
              >
                Max
              </cite>
              <time
                class="review__date"
                datetime="{dayjs(new Date(item.date)).format(\`YYYY-MM-DD\`)}"
              >
                February 18, 2021
              </time>
            </footer>
          </blockquote>
          <div
            class="review__rating"
          >
            3.9
          </div>
        </div>
        <div
          class="review"
        >
          <blockquote
            class="review__quote"
          >
            <p
              class="review__text"
            >
              It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.
            </p>
            <footer
              class="review__details"
            >
              <cite
                class="review__author"
              >
                Christina
              </cite>
              <time
                class="review__date"
                datetime="{dayjs(new Date(item.date)).format(\`YYYY-MM-DD\`)}"
              >
                February 17, 2021
              </time>
            </footer>
          </blockquote>
          <div
            class="review__rating"
          >
            9.7
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MoviePageTabs should render correctly MoviePageTabs should render correctly, when selected Details 1`] = `
<div>
  <div
    class="movie-card__desc"
  >
    <nav
      class="movie-nav movie-card__nav"
    >
      <ul
        class="movie-nav__list"
      >
        <li
          class="movie-nav__item "
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Overview
          </a>
        </li>
        <li
          class="movie-nav__item movie-nav__item--active"
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Details
          </a>
        </li>
        <li
          class="movie-nav__item "
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Reviews
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="movie-card__text movie-card__row"
    >
      <div
        class="movie-card__text-col"
      >
        <p
          class="movie-card__details-item"
        >
          <strong
            class="movie-card__details-name"
          >
            Director
          </strong>
          <span
            class="movie-card__details-value"
          >
            Justin Kurzel
          </span>
        </p>
        <p
          class="movie-card__details-item"
        >
          <strong
            class="movie-card__details-name"
          >
            Starring
          </strong>
          <span
            class="movie-card__details-value"
          >
            Michael Fassbender, Marion Cotillard, Jack Madigan
          </span>
        </p>
      </div>
      <div
        class="movie-card__text-col"
      >
        <p
          class="movie-card__details-item"
        >
          <strong
            class="movie-card__details-name"
          >
            Run Time
          </strong>
          <span
            class="movie-card__details-value"
          >
            1h 53m
          </span>
        </p>
        <p
          class="movie-card__details-item"
        >
          <strong
            class="movie-card__details-name"
          >
            Genre
          </strong>
          <span
            class="movie-card__details-value"
          >
            Drama
          </span>
        </p>
        <p
          class="movie-card__details-item"
        >
          <strong
            class="movie-card__details-name"
          >
            Released
          </strong>
          <span
            class="movie-card__details-value"
          >
            2015
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`MoviePageTabs should render correctly MoviePageTabs should render correctly, when selected Overview 1`] = `
<div>
  <div
    class="movie-card__desc"
  >
    <nav
      class="movie-nav movie-card__nav"
    >
      <ul
        class="movie-nav__list"
      >
        <li
          class="movie-nav__item movie-nav__item--active"
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Overview
          </a>
        </li>
        <li
          class="movie-nav__item "
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Details
          </a>
        </li>
        <li
          class="movie-nav__item "
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Reviews
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="movie-rating"
    >
      <div
        class="movie-rating__score"
      >
        3.3
      </div>
      <p
        class="movie-rating__meta"
      >
        <span
          class="movie-rating__level"
        >
          Normal
        </span>
        <span
          class="movie-rating__count"
        >
          48798
           ratings
        </span>
      </p>
    </div>
    <div
      class="movie-card__text"
    >
      <p>
        Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.
      </p>
      <p
        class="movie-card__director"
      >
        <strong>
          Director: 
          Justin Kurzel
        </strong>
      </p>
      <p
        class="movie-card__starring"
      >
        <strong>
          Starring: 
          Michael Fassbender, Marion Cotillard, Jack Madigan
           and other
        </strong>
      </p>
    </div>
  </div>
</div>
`;

exports[`MoviePageTabs should render correctly MoviePageTabs should render correctly, when selected Reviews 1`] = `
<div>
  <div
    class="movie-card__desc"
  >
    <nav
      class="movie-nav movie-card__nav"
    >
      <ul
        class="movie-nav__list"
      >
        <li
          class="movie-nav__item "
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Overview
          </a>
        </li>
        <li
          class="movie-nav__item "
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Details
          </a>
        </li>
        <li
          class="movie-nav__item movie-nav__item--active"
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Reviews
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="movie-card__reviews movie-card__row"
    >
      <div
        class="movie-card__reviews-col"
      >
        <div
          class="review"
        >
          <blockquote
            class="review__quote"
          >
            <p
              class="review__text"
            >
              Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.
            </p>
            <footer
              class="review__details"
            >
              <cite
                class="review__author"
              >
                Max
              </cite>
              <time
                class="review__date"
                datetime="{dayjs(new Date(item.date)).format(\`YYYY-MM-DD\`)}"
              >
                February 18, 2021
              </time>
            </footer>
          </blockquote>
          <div
            class="review__rating"
          >
            3.9
          </div>
        </div>
        <div
          class="review"
        >
          <blockquote
            class="review__quote"
          >
            <p
              class="review__text"
            >
              It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.
            </p>
            <footer
              class="review__details"
            >
              <cite
                class="review__author"
              >
                Christina
              </cite>
              <time
                class="review__date"
                datetime="{dayjs(new Date(item.date)).format(\`YYYY-MM-DD\`)}"
              >
                February 17, 2021
              </time>
            </footer>
          </blockquote>
          <div
            class="review__rating"
          >
            9.7
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
